tests/nlopt.cpp

```diff
@@ -26,14 +26,22 @@ You should have received copies of the GNU General Public License and the
 GNU Lesser General Public License along with the PaGMO library.  If not,
 see https://www.gnu.org/licenses/. */
 
+#if defined(_MSC_VER)
```

Comment:
would be helpful to have some comments when you add these defines as to remember why they are there in the future